### ISSUE TITLE
Certificate for subdomain with minus sign not added via Webfaction API

### ISSUE THREAD

**Issue by abudarevsky** [will-in-wi/letsencrypt-webfaction#114]
Hello,

I faced situation for domain such as zh-hans.xxxyyyy.net ended up with error:
```
/usr/local/lib/ruby22/ruby/2.2.0/xmlrpc/client.rb:271:in `call': [u'Value must be alphanumeric only'] (XMLRPC::FaultException)
	from /home/user/.letsencrypt_webfaction/gems/gems/letsencrypt_webfaction-2.2.1/lib/letsencrypt_webfaction/webfaction_api_credentials.rb:20:in `call'
	from /home/user/.letsencrypt_webfaction/gems/gems/letsencrypt_webfaction-2.2.1/lib/letsencrypt_webfaction/certificate_installer.rb:20:in `install!'
	from /home/user/.letsencrypt_webfaction/gems/gems/letsencrypt_webfaction-2.2.1/lib/letsencrypt_webfaction/application.rb:32:in `run!'
	from /home/user/.letsencrypt_webfaction/gems/gems/letsencrypt_webfaction-2.2.1/exe/letsencrypt_webfaction:5:in `<top (required)>'
	from /home/user/.letsencrypt_webfaction/gems/bin/letsencrypt_webfaction:23:in `load'
	from /home/user/.letsencrypt_webfaction/gems/bin/letsencrypt_webfaction:23:in `<main>'
```
Is it a bug, api limitation or anything else?

**Comment by abudarevsky** [comment]
Finally resolved with help of support:  --cert_name may not contain spaces, but somehow it affects only domain names with dash.

**Comment by will-in-wi** [comment]
Could you expand on what you mean by it only affects domain names with a dash? Could you give me an example? That sounds like a bug.

**Comment by abudarevsky** [comment]
It is simple - I had few lang subdomains such as ru.mydomain.net, en.mydomain.net, ... One of those was zh-hans.mydomian.net. I ran the script and after awhile I got error above. I suggested that dash was the reason as previously tested script with another group of subdomains without error. When I removed dash I got a warning that cert for zhhans.mydomain.net was not issued which it correct as I don't have one on my list but for the others cert was supplied. I opened a ticket and support suggested that --cert-name may be an issue - I had spaces on my --cert-name, so I replaced spaces with underscore and got cert even for zh-hans.mydomain.net without a complain. I was puzzled how --cert-name related to name of subdomain but I have no time to look Ruby code myself and finally I got cert for everything I needed.
